### PR TITLE
Get, Save Scan

### PR DESCRIPTION
**Pull request by FenrirUnbound**
Implemented the following operations of the In-Memory Database module:

✅ Get
⬜ Remove
✅ Save
✅ Scan

### CODE REVIEW

File: lib/imdb.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * Does not currently support filtering or other query parameters.
+ * @method scanForValues
+ * @param  {String}      tableName         Name of the datastore table
+ * @param  {Object}      queryParams       Parameters to restrict the data set
+ * @param  {Object}      pagination        Pagination options regarding the size of the data set
+ * @param  {Number}      pagination.count  Number of items in the data set
+ * @param  {Number}      pagination.page   The page offset to start with
+ * @return {Array}                         An array of items from the datastore
+ */
+function scanForValues(tableName, queryParams, pagination) {
```
Comment (d2lam) [inline lib/imdb.js]: Where do we take care of query?

Comment (nkatzman) [inline lib/imdb.js]: We haven't implemented that yet

File: lib/imdb.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * @param  {String} tableName Name of the datastore table
+ * @param  {String} keyName   Key of the row to obtain the value from
+ * @return {Object}           The value in the datastore
+ */
+function getValue(tableName, keyName) {
+    if (!tables[tableName]) {
+        return null;
+    }
+
+    return tables[tableName][keyName] || null;
```
Comment (d2lam) [inline lib/imdb.js]: Do we want null or {} ?

Comment (nkatzman) [inline lib/imdb.js]: it's easier to test the truthiness of null. In this scenario, if you ask for a key from the db, I think it'd be better to be returned that object or nothing

Comment (FenrirUnbound) [inline lib/imdb.js]: `null`. If we return `{}`, it's not very clear whether that value is missing or if there really is a `{}` stored there.

**Comment by d2lam** [comment]
Please also include a README

**Comment by jer** [comment]
👍 
bnfzmht - imgur

File: lib/imdb.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ * @param  {Number}      pagination.count  Number of items in the data set
+ * @param  {Number}      pagination.page   The page offset to start with
+ * @return {Array}                         An array of items from the datastore
+ */
+function scanForValues(tableName, queryParams, pagination) {
+    const page = pagination.page - 1;
+    const count = pagination.count;
+    const result = [];
+
+    const keys = Object.keys(tables[tableName]).splice(page * count + 1, count);
```
Comment (d2lam) [inline lib/imdb.js]: Is tables[tableName] an object or an array?

Comment (FenrirUnbound) [inline lib/imdb.js]: A hash

File: test/lib/imdb.test.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            const tableName = 'testTable';
+            const keyPrefix = 'keyPrefix-';
+
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].forEach((value) => {
+                main.set(tableName, keyPrefix + value, value);
+            });
+
+            const result = main.scan(tableName, {}, { page: 2, count: 5 });
+
+            // console.log(result);
```
Comment (FenrirUnbound) [inline test/lib/imdb.test.js]: Remove

File: lib/imdb.js (hunk 3 above)
Comment (d2lam) [inline lib/imdb.js]: Some bug here. Doesn't work on [1,2,3,4,5,6,7] page 3 count 2.

File: lib/imdb.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * Get a value from a specific table
+ * @method getValue
+ * @param  {String} tableName Name of the datastore table
+ * @param  {String} keyName   Key of the row to obtain the value from
+ * @return {Object}           The value in the datastore
+ */
+function getValue(tableName, keyName) {
+    if (!tables[tableName]) {
+        return null;
```
Comment (tkyi) [inline lib/imdb.js]: No error message?

Comment (nkatzman) [inline lib/imdb.js]: Whoever uses this module should be the one to determine what error message to return when there isn't a value available in the datastore

**Comment by tkyi** [comment]
👍 LGTM
giphy-3

**Comment by d2lam** [comment]
👍

**Pull request merged**